Automattic/woocommerce-payments
# Refactor connect account page and tracks

Related to #810. 

**Note**: major part of this PR moved to WC Admin repo due to changed designs. [woocommerce/woocommerce-admin#5248]

#### Changes proposed in this Pull Request

* ~Enable site tracking when onboarding WC Pay.~
* ~Track `Set up` button clicks when tracking disabled.~
* Refactor connect-account-page ui.
* Wrap tracks methods with own utils.

#### Testing instructions

Tests should pass. Set up flow should work as before.

#### Old testing instructions (keep for history)

To test things you'll need ngrok running and [client-example plugin](https://github.com/Automattic/client-example) to manipulate Jetpack connection.

To observer events are tracked: 

* use network tab of the browser dev tools
* enable persistent log to preserve requests on page redirects
* under filter enter .gif
* observe event pixel is sent to wp.com with prop _en= wcadmin_wcpay_connect_account_clicked

Testing sequence:

1. Disable site tracking in WC Settings > Advanced > WooCommerce.com.
2. Disconnect site from Jetpack using client-example plugin.
3. Go to Payments connect page. It's worth clear network tab.
4. Select enable tracking option.
5. Click `Set up` button. Click event should be tracked and page should be redirected to Woo DNA flow (Approve connection screen).
6. Approve connection and check that site tracking is enabled in the settings now (WC Settings > Advanced > WooCommerce.com).
7. Disconnect site again from Jetpack using client-example plugin, but DO NOT disable site tracking.
8. Go to Payments connect page. Tracking option should not be displayed.
9. Repeat steps 5-7. Everything should work the same way.
10. Repeat steps 1 - 6 without selecting enable tracking on connect page.
11. Check that click events are not tracked and site tracking option is **disabled** at (WC Settings > Advanced > WooCommerce.com).


#### Old Screenshots

**Connect page when site tracking is enabled**

<img width="895" alt="wcpay_connect_page_with_enabled_tracking" src="https://user-images.githubusercontent.com/3139099/93765602-d5d55300-fc1d-11ea-9eee-3a0c1ae9d62c.png">


**Connect page when site tracking is disabled**

<img width="892" alt="wcpay_connect_page_with_disabled_tracking" src="https://user-images.githubusercontent.com/3139099/93765618-da9a0700-fc1d-11ea-9866-cd07d55a0501.png">


#### Test coverage notes

UI changes are covered with snapshot tests.
This feature is not covered with tests due to complex interactions with Jetpack and async loading of tracking script.

-------------------
